### PR TITLE
Fixed minor bugs, dealt with filenames having spaces, added cool graphics, etc

### PR DESCRIPTION
**Pull request by whokilleddb**

### CODE REVIEW

File: fake_stream.sh [hunk 1]
```diff
@@ -1,11 +1,55 @@
...
+ ▒ ░    ▒▒   ▓▒█░▒ ▒▒ ▓▒░░ ▒░ ░   ▒ ▒▓▒ ▒ ░  ▒ ░░   ░ ▒▓ ░▒▓░░░ ▒░ ░▒▒   ▓▒█░░ ▒░   ░  ░
+ ░       ▒   ▒▒ ░░ ░▒ ▒░ ░ ░  ░   ░ ░▒  ░ ░    ░      ░▒ ░ ▒░ ░ ░  ░ ▒   ▒▒ ░░  ░      ░
+ ░ ░     ░   ▒   ░ ░░ ░    ░      ░  ░  ░    ░        ░░   ░    ░    ░   ▒   ░      ░   
+             ░  ░░  ░      ░  ░         ░              ░        ░  ░     ░  ░       ░   
+${NONE}"""                                                                                      
+
+# Probing Kernel Modules
 if ! sudo modprobe v4l2loopback exclusive_caps=1; then
-   echo "Unable to probe kernel module."
+   echo -e "${RED}[-] Unable to probe kernel module.${NONE}"
```
Comment (Nibba2018) [inline fake_stream.sh]: Thanks for the PR @whokilleddb .
I will review it tomorrow in the morning.

Just wanted to add one thing. Could you implement a user friendly way to assign webcam device names?(command-line arguments work for me).

webcam names can be assigned by passing the `card_name="webcam name"` argument to the modprobe command.

File: fake_stream.sh [hunk 2]
```diff
@@ -1,11 +1,55 @@
...
    exit ;
 fi
 
-WEBCAMS=$(ls /dev/video*)
-echo $WEBCAMS
-read -p "Choose Webcam ID (last digit):" ID
+
+VIDEO="$@"
+cp "$@" /tmp/video
+
```
Comment (Nibba2018) [inline fake_stream.sh]: Any specific reason why you are copying the video to `/tmp/`?

File: fake_stream.sh [hunk 3]
```diff
@@ -1,11 +1,55 @@
+#!/bin/bash
+
+# function to start fake stream
+startstream()
+{
+      echo -e "${GREEN}${BOLD}[+] Playing $VIDEO On The Stream Pointed By $WEBCAM"
+      ffmpeg -stream_loop -1 -re -i /tmp/video -vcodec rawvideo -pix_fmt yuv420p -threads 0 -f v4l2 $WEBCAM
```
Comment (Nibba2018) [inline fake_stream.sh]: add some comments to breakdown this command.

File: fake_stream.sh [hunk 4]
```diff
@@ -1,11 +1,55 @@
...
+  █████▒▄▄▄       ██ ▄█▀▓█████      ██████ ▄▄▄█████▓ ██▀███  ▓█████ ▄▄▄       ███▄ ▄███▓
+▓██   ▒▒████▄     ██▄█▒ ▓█   ▀    ▒██    ▒ ▓  ██▒ ▓▒▓██ ▒ ██▒▓█   ▀▒████▄    ▓██▒▀█▀ ██▒
+▒████ ░▒██  ▀█▄  ▓███▄░ ▒███      ░ ▓██▄   ▒ ▓██░ ▒░▓██ ░▄█ ▒▒███  ▒██  ▀█▄  ▓██    ▓██░
+░▓█▒  ░░██▄▄▄▄██ ▓██ █▄ ▒▓█  ▄      ▒   ██▒░ ▓██▓ ░ ▒██▀▀█▄  ▒▓█  ▄░██▄▄▄▄██ ▒██    ▒██ 
+░▒█░    ▓█   ▓██▒▒██▒ █▄░▒████▒   ▒██████▒▒  ▒██▒ ░ ░██▓ ▒██▒░▒████▒▓█   ▓██▒▒██▒   ░██▒
+ ▒ ░    ▒▒   ▓▒█░▒ ▒▒ ▓▒░░ ▒░ ░   ▒ ▒▓▒ ▒ ░  ▒ ░░   ░ ▒▓ ░▒▓░░░ ▒░ ░▒▒   ▓▒█░░ ▒░   ░  ░
+ ░       ▒   ▒▒ ░░ ░▒ ▒░ ░ ░  ░   ░ ░▒  ░ ░    ░      ░▒ ░ ▒░ ░ ░  ░ ▒   ▒▒ ░░  ░      ░
+ ░ ░     ░   ▒   ░ ░░ ░    ░      ░  ░  ░    ░        ░░   ░    ░    ░   ▒   ░      ░   
+             ░  ░░  ░      ░  ░         ░              ░        ░  ░     ░  ░       ░   
+${NONE}"""
```
Comment (Nibba2018) [inline fake_stream.sh]: please get rid of the trailing white spaces.

**Review by Nibba2018 — CHANGES_REQUESTED** [review]
There's an issue where modprobe is unable to remove `v4l2loopback` within the cleanup block. Kindly have a look and I have also added some commands below.

`modprobe: FATAL: Module v4l2loopback is in use.`

Apart from that the PR looks ready :+1:

File: fake_stream.sh (hunk 2 above)
Comment (whokilleddb) [reply fake_stream.sh]: In some cases where the Video File's name has a space or special characters in it's name, then using the same in the command directly may throw errors. However, on retrospection, creating a soft link would be more efficient.

**Pull request closed without merging**